pkg/controller/extension/actuator.go

```diff
@@ -136,6 +141,46 @@ func (a *actuator) Restore(context.Context, logr.Logger, *extensionsv1alpha1.Ext
 	return nil
 }
 
+// ReconcileShootInfoConfigMap creates or updates a ConfigMap in the specified Flux namespace in the shoot cluster.
+// The ConfigMap contains information about the Shoot cluster, such as its technical ID which can be used for
+// substitutions in flux kustomizations or helmreleases.
+func ReconcileShootInfoConfigMap(
+	ctx context.Context,
+	log logr.Logger,
+	shootClient client.Client,
+	fluxCfg *fluxv1alpha1.FluxConfig,
+	cluster *extensions.Cluster,
+) error {
+	configMap := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      shootInfoConfigMapName,
+			Namespace: *fluxCfg.Flux.Namespace,
+		},
+	}
+
+	result, err := controllerutil.CreateOrUpdate(ctx, shootClient, configMap, func() error {
+		configMap.Labels = map[string]string{
+			managedByLabelKey: managedByLabelValue,
+		}
+		configMap.Data = map[string]string{
+			"SHOOT_TECHNICAL_ID": cluster.Shoot.Status.TechnicalID,
+			"SHOOT_NAME":         cluster.Shoot.Name,
+			"CLUSTER_NAME":       cluster.Shoot.Name,
+			"SEED_NAME":          *cluster.Shoot.Status.SeedName,
+			"DOMAIN":             *cluster.Shoot.Spec.DNS.Domain,
+		}
+		return nil
+	})
+
+	if err != nil {
+		return fmt.Errorf("failed to create or update %s: %w", shootInfoConfigMapName, err)
+	}
+
+	log.Info("Synced configMap", "Name:", shootInfoConfigMapName, "result", result)
```

Comment:
Can we only log this when something changed?
```suggestion
	if result != controllerutil.OperationResultNone {
		log.Info("Synced shoot info ConfigMap", "configMap", client.ObjectKeyFromObject(configMap))
	}
```